spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/DefaultSftpSessionFactory.java

```diff
@@ -311,6 +324,41 @@ public void setEnableDaemonThread(Boolean enableDaemonThread){
 		this.enableDaemonThread = enableDaemonThread;
 	}
 
+	/**
+	 * Provide a {@link UserInfo} which exposes control over dealing with new keys or key
+	 * changes. As Spring Integration will not normally allow user interaction, the
+	 * implementation must respond to Jsch calls in a suitable way.
+	 * <p>
+	 * Jsch calls {@link UserInfo#promptYesNo(String)} when connecting to an unknown host,
+	 * or when a known host's key has changed (see {@link #setKnownHosts(String)
+	 * unknownHosts}). Generally, it should return false as returning true will accept all
+	 * new keys or key changes.
+	 * <p>
+	 * If no {@link UserInfo} is provided, the behavior is defined by
+	 * {@link #setAllowUnknownKeys(boolean) allowUnknownKeys}.
+	 * <p>
+	 * If {@link #setPassword(String) setPassword} is invoked with a non-null password, it will
+	 * override any password in the supplied {@link UserInfo}.
+	 *
+	 * @param userInfo the UserInfo.
+	 * @see com.jcraft.jsch.Session#setUserInfo(com.jcraft.jsch.UserInfo)
+	 * @since 4.1.7
+	 */
+	public void setUserInfo(UserInfo userInfo) {
+		this.userInfo = userInfo;
+	}
+
+	/**
+	 * When no {@link UserInfo} has been provided, set to true to unconditionally allow
+	 * connecting to an unknown host or when a host's key has changed (see
+	 * {@link #setAllowUnknownKeys(boolean) unknownHosts}). Default false (since 4.2).
```

Comment:
It's just curious what you have had in mind:
![javadocs](https://cloud.githubusercontent.com/assets/1071590/9143010/81ed5916-3d11-11e5-80d0-36b1e2d6cb53.jpg)